hooks/pre_gen_project.py

```diff
@@ -0,0 +1,23 @@
+"""Python module name validation."""
+import logging
+import re
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+def validate_python_module_name():
+    """Check that the module name given by the user is an acceptable format."""
+    module_name = "{{ cookiecutter.module_name }}"
+    # The following regex will only match valid Python module names
+    if not re.match(r"^[_a-zA-Z][_a-zA-Z0-9]+$", module_name):
```

Comment:
I don't think this regex is correct

Reply:
This should fail: 
<img width="309" alt="image" src="https://user-images.githubusercontent.com/12516153/103684201-0bd5e300-4f40-11eb-841a-d068a44a11bd.png">

Reply:
I think it should be this: (no caps or numbers either)
<img width="260" alt="image" src="https://user-images.githubusercontent.com/12516153/103684346-43dd2600-4f40-11eb-9db9-3676a0d70087.png">

Reply:
`^[a-z][_a-z]+$`

Reply:
<img width="804" alt="image" src="https://user-images.githubusercontent.com/12516153/103684400-5b1c1380-4f40-11eb-872e-b9976c8685c5.png">

Reply:
Definitely no caps. It's not super clear from the description, but I'm guess no numbers. Leading underscores seem strange so I'd rather not allow it.

Reply:
I went ahead and removed leading underscores for now and caps. Leaving numbers just because they seem fairly standard in several packages I can think of (`cv2`, for instance), and can be useful in some cases (e.g. `pye57` for python utilities that work with `.e57` files)